invenio_access/__init__.py

```diff
@@ -51,219 +51,316 @@
+transparently by Flask-Security):
 
->>> from invenio_access.models import ActionUsers
->>> db.session.begin(nested=True)
-<sqlalchemy.orm.session.SessionTransaction object ...
->>> db.session.add(ActionUsers.allow(action_index, user=admin))
->>> db.session.commit()
+>>> from invenio_access.utils import get_identity
+>>> alice_identity = get_identity(alice)
+>>> bob_identity = get_identity(bob)
 
-It is then possible to check the permissions:
+Now that we have the identities, we can check if the permissions grants access
+to the identities:
 
->>> permission_index = DynamicPermission(action_index)
->>> anonymous_identity.can(permission_index)
-False
->>> from flask_principal import Identity, RoleNeed, UserNeed
->>> admin_identity = Identity(admin.id)
->>> admin_identity.provides.add(UserNeed(admin.id))
->>> admin_identity.can(permission_index)
+>>> permission.allows(alice_identity)
 True
->>> student_identity = Identity(student.id)
->>> student_identity.provides.add(UserNeed(student.id))
->>> student_identity.can(permission_index)
+>>> permission.allows(bob_identity)
 False
 
+Notice, that we granted access to Alice by assigning her the role ``admin``
+and granting the role permission on the action.
+
+The Flask-Principal API is pretty rich, and there is multiple other ways
+that you can check if a permission grants access to an identity. For instance
+below is another example, but please explore the Flask-Principal API
+documentation for a full reference:
+
+>>> bob_identity.can(permission)
+False
+
+Action parameters
+-----------------
+Above we created an action that did not take any parameters. These actions are
+useful to grant/restrict access to e.g. an entire adminstration interface.
+However, in many cases you need object level permissions, in which case you
+need to use actions with parameters.
+
+Action with parameters are also created with the :py:data:`~.action_facotry`,
+but works a bit different as they take a parameter.
+
+First you create the new action with parameter:
 
-With the command line interface
-+++++++++++++++++++++++++++++++
+>>> ObjectReadAction = action_factory(
+...     'mymodule-object-read', parameter=True)
 
-The permissions can also be assigned using the CLI. For example it is
-possible to allow the action ``open``, which is registered in the
-example application.
+Everytime you create the action, you also need to create an instance of the
+action presenting *any* parameter as done like below:
 
-Let's first intialize the example application which we will use.
+>>> object_read_action_all = ObjectReadAction(None)
 
-.. code-block:: console
+**Granting access to actions with parameters**
 
-    $ cd examples
-    $ export FLASK_APP=app.py
-    $ ./app-setup.sh
+You grant access to actions with parameters in a similar way as for normal
+actions, but you can now grant access to either **all** objects:
 
-Let's create a user.
+>>> db.session.add(ActionRoles.allow(
+...     object_read_action_all, role=admin))
```

Comment:
too much complexity here. Is it really required, or we can just do

```python
db.session.add(ActionRoles.allow(ObjectReadAction(), role=admin))
```

?
Then, creating the `object_read_action_all` is just a good practice and we may remove it from here to simplify (?)